// config/models.js
var admin = "trunganh";
var user = "andrew";
var pass = "28111994";
var server = "127.0.0.1";
var port = "27017";
var dbname = "ChatApp-ByAndrew";
// var opt = {
// 	user : adminuser,
// 	pass : adminpass,
// 	auth : {
// 		authdb: 'admin'
// 	}
// }
var connString = "mongodb://localhost:27017/" + dbname;

var mongoose = require('mongoose');

var Schema = mongoose.Schema;

var userSchema = mongoose.Schema({
	token : String,
	email : String,
	username : String,
	hashed_password : String,
	salt : String,  
    gender : String,
    avatar : String,
    state : String,
    cover_image : String
});

var createDefaultCollection = function(username){
	var variable = {};
	var img_username = "image_" + username;
	var chat_username = "chat_" + username;
	var contact_username = "contact_" + username;
	var profile_username = "profile_" + username;

	variable[img_username] = mongoose.Schema({
		username : String,
		image : String,
		from_date : Date,
		thru_date : Date
	})
	// var "image_" + username = mongoose.Schema({
	// 	username : String,
	// 	image : String,
	// 	from_date : Date,
	// 	thru_date : Date
	// });
	variable[chat_username] = mongoose.Schema({
		username : String,
		guest : String,
		message : String,
		from_date : Date
	});
	// var "chat_" + username = mongoose.Schema({
	// 	username : String,
	// 	guest : String,
	// 	message : String,
	// 	from_date : Date
	// });
	variable[contact_username] = mongoose.Schema({
		username : String,
		contact : String,
		state : String
	});
	// var "contact_" + username = mongoose.Schema({
	// 	username : String,
	// 	contact : String,
	// 	state : String
	// });
	variable[profile_username] = mongoose.Schema({
		username : String,
		gender : String,
		age : Number,
		school_name : String,
		company_name : String,
		favorite : String,
		relationship_status : String,
		address : String
	});

	module.exports['image_' + username] = mongoose.model('image_' + username, variable[img_username]);
	module.exports['chat_' + username] = mongoose.model('chat_' + username, variable[chat_username]);
	module.exports['contact_' + username] = mongoose.model('contact_' + username, variable[contact_username]);
	module.exports['profile_' + username] = mongoose.model('profile_' + username, variable[profile_username]);

	return module;
}

mongoose.connect(connString);
var db = mongoose.connection;
db.on('error', console.error.bind(console, 'connection error:'));
db.once('open', function(){
})
module.exports = {
	user : mongoose.model('users', userSchema),
	createDefaultCollection : createDefaultCollection
};
